tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -665,9 +665,10 @@ _ when x.ToBoolean() is { } boolean => boolean,
                              .WithKeys(ConfigurationKeys.IpHeaderEnabled)
                              .AsBool(false);
 
+            // DSM is now enabled by default in non-serverless environments
             _isDataStreamsMonitoringEnabled = config
                                             .WithKeys(ConfigurationKeys.DataStreamsMonitoring.Enabled)
-                                            .AsBool(false);
+                                            .AsBool(!EnvironmentHelpers.IsServerlessEnvironment());
```

Comment:
Note that this means it _will_ be enabled when using [the AAS extension](https://github.com/DataDog/datadog-aas-extension). This is pinned to using `trace-agent` `v7.60.1` and does _not_ include the full agent - does that include the required endpoints for DSM?

Reply:
Related: I'm currently doing some refactoring around how we detect Azure App Service and Azure Functions in #7256 because we will soon support instrumenting without the AAS Site Extension.